pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -37,6 +37,10 @@ import (
 // maxUpdateRetries is the maximum number of retries used for update conflict resolution prior to failure
 const maxUpdateRetries = 10
 
+// StatefulSetPodNameLabel is label added to each Pod in a StatefulSet. Its purpose is to allow Services to select
+// individual Pods in the StatefulSet.
+const StatefulSetPodNameLabel = "statefulset.kubernetes.io/podName"
```

Comment:
Most workload controllers use `xxx-xxx-xxx` format as label key (e.g. pod-template-hash, controller-revision-hash). 

Deployment hash label was named `pod-template-hash` instead of something like `deployment.kubernetes.io/podTemplateHash`, because the latter is lengthy and more suitable as annotations -- we expect users to type label name (e.g. `kubectl get name -l`) more frequently than annotations. 

This is not documented AFAIK. The convention may have changed.